### DISCUSSION TITLE
Not mapping data object values in method parameter

### DISCUSSION THREAD

**Discussion by oskarjf** [CoreWCF/CoreWCF#1252] [Q&A] [answered]
I've got a method marked with [OperationContract] that has a data object as parameter. For some reason this parameter object is not initialized with the data sent with the POST operation.
The endpoint has configured a basicHttpBinding.
The data object has multiple constructors.

Any idea as to where I should be looking?

I've tried to decorate the data object class with [DataContract] and it's members with [DataMember].
I'm using CoreWCF 1.4.1

**Comment by oskarjf** [comment]
I'm testing this with Postman.  Most endpoints work.
I've tried to simplify the object (DTO) to just 
`    [DataContract]
    public class ActionSimpleDto
    {
        [DataMember]
        public int Id { get; set; }`

My method:
`        public int CreateOrUpdateActionSimple(ActionSimpleDto dto, ICurrentInfo currentInfo = null)
        {
            return dto.Id;
        }
`

My Postman XML:
`  <soap:Body>
    <CreateOrUpdateActionSimple xmlns="http://tempuri.org/">
    <dto>
         <id>258260</id>
    </dto>
    </CreateOrUpdateActionSimple>
  </soap:Body>
`
I've tried with <id> and <Id> with the same result.

When debugging in VS I can see that the Id of the item (dto) is always 0.  Any ideas?

Btw, I'm using a custom IOperationInvoker and I can see that InvokeAsync's inputs parameter has this parameter with it's default values.

**Reply by mconnew** [reply]
I suspect this is a casing problem. The property is called `Id`, but the xml is `id`. Try changing your class to this:
```csharp
[DataContract]
public class ActionSimpleDto 
{ 
    [DataMember(Name = "id")]
    public int Id { get; set; }
}
```

**Reply by oskarjf** [reply]
I've added name to the property as suggested but the value still doesn't get set.  I've tried different casing in the XML request but with the same result.

**Reply by oskarjf** [reply]
If I add another parameter to my method with another object that doesn't get set either. If I add a parameter that is just int then that value gets set.

`
    [DataContract]
    public class OJFTestClass
    {
        [DataMember(Name = "name")]
        public string Name { get; set; }
    }
`

`public int CreateOrUpdateActionSimple(ActionSimpleDto dto, OJFTestClass ojf, int id = 0)`

`
<dto>
         <id>258260</id>
    </dto>
    <ojf>
        <name>OJF</name>
    </ojf>
    <id>1233</id>
`

ojf.Name is null and dto.Id is 0.  The id parameter gets the value 1233 as sent in the request.

Could I be missing some setting in my project or something?

**Reply by oskarjf** [reply]
I've only installed the CoreWCF.Primitives NuGet package.  Not sure when I should be installing other CoreWCF packages like CoreWCF.Http and CoreWCF.NetTcp

**Reply by oskarjf** [reply]
I might also mention that this service was upgraded from WCF/.Net Framework. to CoreWCF/.Net 7

**Reply by oskarjf** [reply]
Could this be some XmlSerializer issue? I see that mentioned a lot in similar WCF problems.

**Reply by oskarjf** [reply]
My problem sounds similar to this question at StackOverflow:
https://stackoverflow.com/questions/13178958/sending-complex-type-as-a-parameter-in-soap-message

**Reply by mconnew** [reply]
When using `[DataContract]` it uses `DataContractSerializer` and not `XmlSerializer`. Are you able to use the same contract with a WCF client (using System.ServiceModel.* packages) running on .NET 7? That will use the same implementation of `DataContractSerializer`. If you can then look at what is sent, it might provide some clues as to the problem. We have explicit tests for contracts which look similar to this, but they are all in-process tests so if there's a problem with DCS serializing slightly different XML but that problem also allows it to deserialize it, we won't pick it up in the CoreWCF repo. The WCF client repo does run the .NET client against services running in .NET Framework so I would have expected that to pick up an issue. And the runtime DCS tests do validate against expected XML output so also would expect an issue to be picked up there.  
You can use [classic fiddler](https://www.telerik.com/download/fiddler) to capture the request. The easiest way is to configure the client to use an actual hostname (using `localhost` bypasses a proxy server by default) and configuring fiddler to be the system proxy.

**Reply by oskarjf** [reply, marked as answer]
Sorry @mconnew ! It's been too long since I worked last with SOAP. Of course the issue was my XML request where I hadn't put the namespace of the data object on the XML element.  Should have used Soap UI earlier.
Thanks
